src/Pages/login.jsx

```diff
@@ -0,0 +1,111 @@
+import { event } from "jquery";
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+export default () => {
+  const navigate = useNavigate();
+  const [userName, setUserName] = useState(null);
+  const [password, setPassword] = useState(null);
+
+  const [isUsernameActivated, setUsernameActivated] = useState(false);
+  const [isPasswordActivated, setPasswordActivated] = useState(false);
+
+  const [unalert, setUnAlert] = useState(null);
+  const [pwalert, setPwAlert] = useState(null);
+
+  function handleUsername(event){
+    setUserName(event.target.value);
+    
+    checkUsername(userName);
+  
+  }
+
+  function handleUnBlur() {
+    setUsernameActivated(true)
+    checkUsername(userName)
```

Comment:
It's good that you've thought of code separation/chunking here. But we should always do separations when they make sense. In this case, the function logic is not too complex, neither is the separated code, and there is no code duplication scenarios associated with the separation. It would be easier to have all the logic in one place rather than having to look at two different places to understand the input logic.